src/js/stores/document.js

```diff
@@ -169,7 +169,17 @@ define(function (require, exports, module) {
 
             this._openDocuments[nextDocument.id] = nextDocument;
 
-            if (!suppressChange) {
+            if (suppressChange) {
+                return;
+            }
+
+            // If some selected layer remains uninitialized, a new document will
+            // come along shortly. Wait until then to trigger the change event.
+            var initialized = nextDocument.layers.selected.every(function (layer) {
```

Comment:
so far this is the only thing that creeps me out.  Is it not possible to wait on all the layers being initialized before calling setDocument?  Is it somehow not desirable to do so?